index.js

```diff
@@ -16,8 +17,7 @@ function underPressure (fastify, opts, next) {
 
   var heapUsed = 0
   var rssBytes = 0
-  var eventLoopDelay = 0
-  var lastCheck = now()
+  var loopSampler = loopDelay()
   const timer = setInterval(updateMemoryUsage, sampleInterval)
```

Comment:
sampleInterval should be renamed to "memorySampleInterval", as the sample interval in event-loop-delay cannot be configured: https://github.com/mafintosh/event-loop-delay/blob/master/native/binding.c#L5.

Also, there is no reason to check memory every 5ms, we can definitely check it every second or so.

Reply:
> No you should not be doing this. Why are you doing it?

In docs it says that it is recording cumulative delay since the creation of the sampler. So if I am thinking correctly it will always increase and will eventually pass the limit, then stay above the limit. Thats why I am a bit confused.

Reply:
>No you should not be doing this. Why are you doing it?

In docs it says that it is recording cumulative delay since the creation of the sampler. So if I am thinking correctly it will always increase and will eventually pass the limit, then stay above the limit. Thats why I am a bit confused.

Reply:
> In docs it says that it is recording cumulative delay since the creation of the sampler. So if I am thinking correctly it will always increase and will eventually pass the limit, then stay above the limit. Thats why I am a bit confused.

Oh, true. Then you should destroy the previous loopSampler, otherwise you'll be leaking memory.

Reply:
Something still feels off though event with resetting. Since this is cumulative, on a long enough timeline it is guaranteed to reach the delay limit producing an unnecessary 503.

Reply:
I don't think this PR is doable without some changes to event-loop-delay https://github.com/mafintosh/event-loop-delay/issues/2.